source/faq.txt

```diff
@@ -4,14 +4,99 @@
 FAQ
 ===
 
-.. default-domain:: mongodb
-
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 2
+   :depth: 1
    :class: singlecol
 
+This page contains frequently asked questions and their corresponding answers.
+
+.. tip::
+
+   If you can't find an answer to your problem on this page,
+   see the :ref:`csharp-issues-help` page for next steps and more
+   resources.
+
+.. _csharp-faq-connection-pool:
+
+How Does Connection Pooling Work in the {+driver-short+}?
+---------------------------------------------------------
+
+Every ``MongoClient`` instance has a built-in connection pool for each server
+in your MongoDB topology. Connection pools open sockets on demand to
+support concurrent MongoDB operations in your multi-threaded application.
+
+The maximum size of each connection pool is set by the ``MaxConnectionPoolSize`` option, which
+defaults to ``100``. If the number of in-use connections to a server reaches
+the value of ``MaxConnectionPoolSize``, the next request to that server will wait
+until a connection becomes available. 
+
+The driver has a wait queue that limits the number of threads that can
+wait for a connection. The size of the wait queue is determined by the
+``WaitQueueMultiple`` option, which defaults to ``5``. The wait queue
+size is ``WaitQueueMultiple`` x ``MaxConnectionPoolSize``, which is ``500`` by
+default. You can also set the wait queue size by specifying the
+``WaitQueueSize`` option, which overrides the other settings. However,
+we do not recommend changing the wait queue size from the default.
```

Comment:
I feel like putting things in admonitions in the FAQ feels unnecessary, since this content is an "answer," not a guide or anything. Maybe I'm mistaken in feeling that way -- lmk if you think I should do it anyways.

Reply:
That's fair enough.